Installation issue: minimap2 (for aarch64)

When I build minimap2 on my aarch64 platform, build error is reported:

### Steps to reproduce the issue

```console
$ spack install -v minimap2
[+] /home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/libiconv-1.16-l55npm3aywt3ht3trtormu2flaeejhym
[+] /home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/libbsd-0.10.0-f7zri6ars4f45jxwxbwee523uunsc5tv
[+] /home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/pkgconf-1.6.3-xwwcou2uvone45j36scf7atvesbadmah
[+] /home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/xz-5.2.5-n5spgwkq2ixqb3ubruaxdifholuuuefg
[+] /home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/zlib-1.2.11-r26fdvv4oowwlm4yozeoozfvg6kkkeoa
[+] /home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/libffi-3.3-oa2qbxkmldfshfwagwdgbirvlp4a2jbk
[+] /home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/tar-1.32-mmql2dzt3fuxp3bd6jswhxxcvhseddfw
[+] /home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/diffutils-3.7-k4qj43xag5ztbnv5nzkcexzo73et7yub
[+] /home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/expat-2.2.9-7pnvot7flpdcaq4a5eggnw3quyqqmtjr
[+] /home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/ncurses-6.2-5zs46oqspomfxll4jasxilx3xnw24ylc
[+] /home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/libxml2-2.9.10-n3yjvwrs62zscl6dgkgf6hxyofhb245h
[+] /home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/bzip2-1.0.8-3z4cm7torpanvuhuihxyb27zr3mjkarn
[+] /home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/readline-8.0-o72lkv3gc22wle6le25ijxakj522uf7i
[+] /home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/gettext-0.20.2-xy3rmphnbgssrolk322q3bhtcl73sfq2
[+] /home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/gdbm-1.18.1-bskc6oghdwgcgxhv6n233xqmfmgqvtqc
[+] /home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/sqlite-3.31.1-oglqqnuumbcfbvyr5226uay5h4w4jbxu
[+] /home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/perl-5.30.2-6ydyb2rrgcp575pbiqrquc4ezig2362m
[+] /home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/openssl-1.1.1g-ku5v5n2vxpjzaz6e4h7mavavabf7pvuw
[+] /home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/python-3.7.7-s5nnfu3nlfzwua6ep5oahgofpmwrcght
[+] /home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/py-setuptools-46.1.3-ejav3ouazxfgndpvphv4k2bdy4dxqipx
[+] /home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/py-cython-0.29.16-gnbtewtlmqvwy547av5zgjhxpjssd3z2
==> 1549: Installing minimap2
==> Using cached archive: /home/xiaojun/spack/var/spack/cache/_source-cache/archive/90/9088b785bb0c33488ca3a27c8994648ce21a8be54cb117f5ecee26343facd03b.tar.bz2
==> Staging archive: /tmp/root/spack-stage/spack-stage-minimap2-2.14-st347n5vburvgv2qijtmmsfa2w2oteno/minimap2-2.14.tar.bz2
==> Created stage in /tmp/root/spack-stage/spack-stage-minimap2-2.14-st347n5vburvgv2qijtmmsfa2w2oteno
==> No patches needed for minimap2
==> 1549: minimap2: Building minimap2 [PythonPackage]
==> 1549: minimap2: Executing phase: 'build'
==> [2020-06-16-06:18:33.722203, 2237] '/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/python-3.7.7-s5nnfu3nlfzwua6ep5oahgofpmwrcght/bin/python3.7' '-s' 'setup.py' '--no-user-cfg' 'build'
running build
running build_ext
cythoning python/mappy.pyx to python/mappy.c
building 'mappy' extension
creating build
creating build/temp.linux-aarch64-3.7
creating build/temp.linux-aarch64-3.7/python
/home/xiaojun/spack/lib/spack/env/gcc/gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -fPIC -I. -Isse2neon/ -I/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/python-3.7.7-s5nnfu3nlfzwua6ep5oahgofpmwrcght/include/python3.7m -c python/mappy.c -o build/temp.linux-aarch64-3.7/python/mappy.o -DHAVE_KALLOC -ftree-vectorize -DKSW_SSE2_ONLY -D__SSE2__
/home/xiaojun/spack/lib/spack/env/gcc/gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -fPIC -I. -Isse2neon/ -I/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/python-3.7.7-s5nnfu3nlfzwua6ep5oahgofpmwrcght/include/python3.7m -c align.c -o build/temp.linux-aarch64-3.7/align.o -DHAVE_KALLOC -ftree-vectorize -DKSW_SSE2_ONLY -D__SSE2__
align.c: In function 'mm_get_hplen_back':
align.c:326:29: warning: comparison of integer expressions of different signedness: 'uint32_t' {aka 'unsigned int'} and 'int' [-Wsign-compare]
   if (mm_seq4_get(mi->S, i) != c) break;
                             ^~
align.c: In function 'mm_seed_ext_score':
align.c:513:49: warning: operand of ?: changes signedness from 'int' to 'uint32_t' {aka 'unsigned int'} due to unsignedness of other operand [-Wsign-compare]
  re = re + ext_len < (int32_t)mi->seq[rid].len? re + ext_len : mi->seq[rid].len;
                                                 ^~~~~~~~~~~~
align.c: In function 'mm_align1':
align.c:600:45: warning: operand of ?: changes signedness from 'int32_t' {aka 'int'} to 'uint32_t' {aka 'unsigned int'} due to unsignedness of other operand [-Wsign-compare]
   re0 = re + l < (int32_t)mi->seq[rid].len? re + l : mi->seq[rid].len;
                                             ^~~~~~
align.c:650:44: warning: operand of ?: changes signedness from 'int32_t' {aka 'int'} to 'uint32_t' {aka 'unsigned int'} due to unsignedness of other operand [-Wsign-compare]
    l = l < (int32_t)mi->seq[rid].len - re? l : mi->seq[rid].len - re;
                                            ^
/home/xiaojun/spack/lib/spack/env/gcc/gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -fPIC -I. -Isse2neon/ -I/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/python-3.7.7-s5nnfu3nlfzwua6ep5oahgofpmwrcght/include/python3.7m -c bseq.c -o build/temp.linux-aarch64-3.7/bseq.o -DHAVE_KALLOC -ftree-vectorize -DKSW_SSE2_ONLY -D__SSE2__
/home/xiaojun/spack/lib/spack/env/gcc/gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -fPIC -I. -Isse2neon/ -I/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/python-3.7.7-s5nnfu3nlfzwua6ep5oahgofpmwrcght/include/python3.7m -c chain.c -o build/temp.linux-aarch64-3.7/chain.o -DHAVE_KALLOC -ftree-vectorize -DKSW_SSE2_ONLY -D__SSE2__
/home/xiaojun/spack/lib/spack/env/gcc/gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -fPIC -I. -Isse2neon/ -I/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/python-3.7.7-s5nnfu3nlfzwua6ep5oahgofpmwrcght/include/python3.7m -c format.c -o build/temp.linux-aarch64-3.7/format.o -DHAVE_KALLOC -ftree-vectorize -DKSW_SSE2_ONLY -D__SSE2__
/home/xiaojun/spack/lib/spack/env/gcc/gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -fPIC -I. -Isse2neon/ -I/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/python-3.7.7-s5nnfu3nlfzwua6ep5oahgofpmwrcght/include/python3.7m -c hit.c -o build/temp.linux-aarch64-3.7/hit.o -DHAVE_KALLOC -ftree-vectorize -DKSW_SSE2_ONLY -D__SSE2__
hit.c: In function 'mm_hit_sort':
hit.c:171:48: warning: variable 'no_cigar' set but not used [-Wunused-but-set-variable]
  int32_t i, n_aux, n = *n_regs, has_cigar = 0, no_cigar = 0;
                                                ^~~~~~~~
hit.c:171:33: warning: variable 'has_cigar' set but not used [-Wunused-but-set-variable]
  int32_t i, n_aux, n = *n_regs, has_cigar = 0, no_cigar = 0;
                                 ^~~~~~~~~
hit.c: In function 'mm_join_long':
hit.c:324:40: warning: operand of ?: changes signedness from 'int' to 'uint64_t' {aka 'long unsigned int'} due to unsignedness of other operand [-Wsign-compare]
   max_gap = a0e->x + max_gap > a1s->x? max_gap : a1s->x - a0e->x;
                                        ^~~~~~~
hit.c:325:40: warning: operand of ?: changes signedness from 'int' to 'uint64_t' {aka 'long unsigned int'} due to unsignedness of other operand [-Wsign-compare]
   min_gap = a0e->x + min_gap < a1s->x? min_gap : a1s->x - a0e->x;
                                        ^~~~~~~
/home/xiaojun/spack/lib/spack/env/gcc/gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -fPIC -I. -Isse2neon/ -I/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/python-3.7.7-s5nnfu3nlfzwua6ep5oahgofpmwrcght/include/python3.7m -c index.c -o build/temp.linux-aarch64-3.7/index.o -DHAVE_KALLOC -ftree-vectorize -DKSW_SSE2_ONLY -D__SSE2__
index.c: In function 'mm_idx_name2id':
index.c:134:25: warning: operand of ?: changes signedness from 'int' to 'uint32_t' {aka 'unsigned int'} due to unsignedness of other operand [-Wsign-compare]
  return k == kh_end(h)? -1 : kh_val(h, k);
                         ^~
index.c: In function 'mm_idx_gen':
index.c:343:63: warning: operand of ?: changes signedness from 'int' to 'uint64_t' {aka 'long unsigned int'} due to unsignedness of other operand [-Wsign-compare]
  pl.mini_batch_size = (uint64_t)mini_batch_size < batch_size? mini_batch_size : batch_size;
                                                               ^~~~~~~~~~~~~~~
index.c: In function 'mm_idx_dump':
index.c:443:16: warning: comparison of integer expressions of different signedness: 'uint32_t' {aka 'unsigned int'} and 'int' [-Wsign-compare]
  for (i = 0; i < 1<<mi->b; ++i) {
                ^
index.c: In function 'mm_idx_load':
index.c:490:16: warning: comparison of integer expressions of different signedness: 'uint32_t' {aka 'unsigned int'} and 'int' [-Wsign-compare]
  for (i = 0; i < 1<<mi->b; ++i) {
                ^
/home/xiaojun/spack/lib/spack/env/gcc/gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -fPIC -I. -Isse2neon/ -I/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/python-3.7.7-s5nnfu3nlfzwua6ep5oahgofpmwrcght/include/python3.7m -c pe.c -o build/temp.linux-aarch64-3.7/pe.o -DHAVE_KALLOC -ftree-vectorize -DKSW_SSE2_ONLY -D__SSE2__
/home/xiaojun/spack/lib/spack/env/gcc/gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -fPIC -I. -Isse2neon/ -I/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/python-3.7.7-s5nnfu3nlfzwua6ep5oahgofpmwrcght/include/python3.7m -c options.c -o build/temp.linux-aarch64-3.7/options.o -DHAVE_KALLOC -ftree-vectorize -DKSW_SSE2_ONLY -D__SSE2__
/home/xiaojun/spack/lib/spack/env/gcc/gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -fPIC -I. -Isse2neon/ -I/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/python-3.7.7-s5nnfu3nlfzwua6ep5oahgofpmwrcght/include/python3.7m -c ksw2_extd2_sse.c -o build/temp.linux-aarch64-3.7/ksw2_extd2_sse.o -DHAVE_KALLOC -ftree-vectorize -DKSW_SSE2_ONLY -D__SSE2__
/home/xiaojun/spack/lib/spack/env/gcc/gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -fPIC -I. -Isse2neon/ -I/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/python-3.7.7-s5nnfu3nlfzwua6ep5oahgofpmwrcght/include/python3.7m -c ksw2_exts2_sse.c -o build/temp.linux-aarch64-3.7/ksw2_exts2_sse.o -DHAVE_KALLOC -ftree-vectorize -DKSW_SSE2_ONLY -D__SSE2__
/home/xiaojun/spack/lib/spack/env/gcc/gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -fPIC -I. -Isse2neon/ -I/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/python-3.7.7-s5nnfu3nlfzwua6ep5oahgofpmwrcght/include/python3.7m -c ksw2_extz2_sse.c -o build/temp.linux-aarch64-3.7/ksw2_extz2_sse.o -DHAVE_KALLOC -ftree-vectorize -DKSW_SSE2_ONLY -D__SSE2__
/home/xiaojun/spack/lib/spack/env/gcc/gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -fPIC -I. -Isse2neon/ -I/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/python-3.7.7-s5nnfu3nlfzwua6ep5oahgofpmwrcght/include/python3.7m -c ksw2_ll_sse.c -o build/temp.linux-aarch64-3.7/ksw2_ll_sse.o -DHAVE_KALLOC -ftree-vectorize -DKSW_SSE2_ONLY -D__SSE2__
/home/xiaojun/spack/lib/spack/env/gcc/gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -fPIC -I. -Isse2neon/ -I/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/python-3.7.7-s5nnfu3nlfzwua6ep5oahgofpmwrcght/include/python3.7m -c kalloc.c -o build/temp.linux-aarch64-3.7/kalloc.o -DHAVE_KALLOC -ftree-vectorize -DKSW_SSE2_ONLY -D__SSE2__
/home/xiaojun/spack/lib/spack/env/gcc/gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -fPIC -I. -Isse2neon/ -I/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/python-3.7.7-s5nnfu3nlfzwua6ep5oahgofpmwrcght/include/python3.7m -c kthread.c -o build/temp.linux-aarch64-3.7/kthread.o -DHAVE_KALLOC -ftree-vectorize -DKSW_SSE2_ONLY -D__SSE2__
/home/xiaojun/spack/lib/spack/env/gcc/gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -fPIC -I. -Isse2neon/ -I/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/python-3.7.7-s5nnfu3nlfzwua6ep5oahgofpmwrcght/include/python3.7m -c map.c -o build/temp.linux-aarch64-3.7/map.o -DHAVE_KALLOC -ftree-vectorize -DKSW_SSE2_ONLY -D__SSE2__
map.c: In function 'mm_dust_minier':
map.c:52:42: warning: operand of ?: changes signedness from 'int32_t' {aka 'int'} to 'uint64_t' {aka 'const long unsigned int'} due to unsignedness of other operand [-Wsign-compare]
     int ss = s > (int32_t)(dreg[v]>>32)? s : dreg[v]>>32;
                                          ^
map.c:53:36: warning: operand of ?: changes signedness from 'int32_t' {aka 'int'} to 'unsigned int' due to unsignedness of other operand [-Wsign-compare]
     int ee = e < (int32_t)dreg[v]? e : (uint32_t)dreg[v];
                                    ^
/home/xiaojun/spack/lib/spack/env/gcc/gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -fPIC -I. -Isse2neon/ -I/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/python-3.7.7-s5nnfu3nlfzwua6ep5oahgofpmwrcght/include/python3.7m -c misc.c -o build/temp.linux-aarch64-3.7/misc.o -DHAVE_KALLOC -ftree-vectorize -DKSW_SSE2_ONLY -D__SSE2__
/home/xiaojun/spack/lib/spack/env/gcc/gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -fPIC -I. -Isse2neon/ -I/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/python-3.7.7-s5nnfu3nlfzwua6ep5oahgofpmwrcght/include/python3.7m -c sdust.c -o build/temp.linux-aarch64-3.7/sdust.o -DHAVE_KALLOC -ftree-vectorize -DKSW_SSE2_ONLY -D__SSE2__
/home/xiaojun/spack/lib/spack/env/gcc/gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -fPIC -I. -Isse2neon/ -I/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/python-3.7.7-s5nnfu3nlfzwua6ep5oahgofpmwrcght/include/python3.7m -c sketch.c -o build/temp.linux-aarch64-3.7/sketch.o -DHAVE_KALLOC -ftree-vectorize -DKSW_SSE2_ONLY -D__SSE2__
/home/xiaojun/spack/lib/spack/env/gcc/gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -fPIC -I. -Isse2neon/ -I/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/python-3.7.7-s5nnfu3nlfzwua6ep5oahgofpmwrcght/include/python3.7m -c esterr.c -o build/temp.linux-aarch64-3.7/esterr.o -DHAVE_KALLOC -ftree-vectorize -DKSW_SSE2_ONLY -D__SSE2__
/home/xiaojun/spack/lib/spack/env/gcc/gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -fPIC -I. -Isse2neon/ -I/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/python-3.7.7-s5nnfu3nlfzwua6ep5oahgofpmwrcght/include/python3.7m -c splitidx.c -o build/temp.linux-aarch64-3.7/splitidx.o -DHAVE_KALLOC -ftree-vectorize -DKSW_SSE2_ONLY -D__SSE2__
creating build/lib.linux-aarch64-3.7
/usr/bin/gcc -pthread -shared -L/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/gettext-0.20.2-xy3rmphnbgssrolk322q3bhtcl73sfq2/lib -L/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/readline-8.0-o72lkv3gc22wle6le25ijxakj522uf7i/lib -L/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/ncurses-6.2-5zs46oqspomfxll4jasxilx3xnw24ylc/lib -L/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/openssl-1.1.1g-ku5v5n2vxpjzaz6e4h7mavavabf7pvuw/lib -L/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/sqlite-3.31.1-oglqqnuumbcfbvyr5226uay5h4w4jbxu/lib -L/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/gdbm-1.18.1-bskc6oghdwgcgxhv6n233xqmfmgqvtqc/lib -L/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/zlib-1.2.11-r26fdvv4oowwlm4yozeoozfvg6kkkeoa/lib -L/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/bzip2-1.0.8-3z4cm7torpanvuhuihxyb27zr3mjkarn/lib -L/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/xz-5.2.5-n5spgwkq2ixqb3ubruaxdifholuuuefg/lib -L/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/expat-2.2.9-7pnvot7flpdcaq4a5eggnw3quyqqmtjr/lib -L/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/libffi-3.3-oa2qbxkmldfshfwagwdgbirvlp4a2jbk/lib64 build/temp.linux-aarch64-3.7/python/mappy.o build/temp.linux-aarch64-3.7/align.o build/temp.linux-aarch64-3.7/bseq.o build/temp.linux-aarch64-3.7/chain.o build/temp.linux-aarch64-3.7/format.o build/temp.linux-aarch64-3.7/hit.o build/temp.linux-aarch64-3.7/index.o build/temp.linux-aarch64-3.7/pe.o build/temp.linux-aarch64-3.7/options.o build/temp.linux-aarch64-3.7/ksw2_extd2_sse.o build/temp.linux-aarch64-3.7/ksw2_exts2_sse.o build/temp.linux-aarch64-3.7/ksw2_extz2_sse.o build/temp.linux-aarch64-3.7/ksw2_ll_sse.o build/temp.linux-aarch64-3.7/kalloc.o build/temp.linux-aarch64-3.7/kthread.o build/temp.linux-aarch64-3.7/map.o build/temp.linux-aarch64-3.7/misc.o build/temp.linux-aarch64-3.7/sdust.o build/temp.linux-aarch64-3.7/sketch.o build/temp.linux-aarch64-3.7/esterr.o build/temp.linux-aarch64-3.7/splitidx.o -L/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/python-3.7.7-s5nnfu3nlfzwua6ep5oahgofpmwrcght/lib -lz -lm -lpthread -lpython3.7m -o build/lib.linux-aarch64-3.7/mappy.cpython-37m-aarch64-linux-gnu.so
running build_scripts
creating build/scripts-3.7
copying and adjusting python/minimap2.py -> build/scripts-3.7
changing mode of build/scripts-3.7/minimap2.py from 644 to 755
/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/py-cython-0.29.16-gnbtewtlmqvwy547av5zgjhxpjssd3z2/lib/python3.7/site-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /tmp/root/spack-stage/spack-stage-minimap2-2.14-st347n5vburvgv2qijtmmsfa2w2oteno/spack-src/python/mappy.pyx
  tree = Parsing.p_module(s, pxd, full_module_name)
==> 1549: minimap2: Executing phase: 'install'
==> [2020-06-16-06:18:54.387444, 2237] '/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/python-3.7.7-s5nnfu3nlfzwua6ep5oahgofpmwrcght/bin/python3.7' '-s' 'setup.py' '--no-user-cfg' 'install' '--prefix=/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/minimap2-2.14-st347n5vburvgv2qijtmmsfa2w2oteno'
running install
running bdist_egg
running egg_info
creating mappy.egg-info
writing mappy.egg-info/PKG-INFO
writing dependency_links to mappy.egg-info/dependency_links.txt
writing top-level names to mappy.egg-info/top_level.txt
writing manifest file 'mappy.egg-info/SOURCES.txt'
reading manifest file 'mappy.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'mappy.egg-info/SOURCES.txt'
installing library code to build/bdist.linux-aarch64/egg
running install_lib
running build_ext
skipping 'python/mappy.c' Cython extension (up-to-date)
creating build/bdist.linux-aarch64
creating build/bdist.linux-aarch64/egg
copying build/lib.linux-aarch64-3.7/mappy.cpython-37m-aarch64-linux-gnu.so -> build/bdist.linux-aarch64/egg
creating stub loader for mappy.cpython-37m-aarch64-linux-gnu.so
byte-compiling build/bdist.linux-aarch64/egg/mappy.py to mappy.cpython-37.pyc
creating build/bdist.linux-aarch64/egg/EGG-INFO
installing scripts to build/bdist.linux-aarch64/egg/EGG-INFO/scripts
running install_scripts
running build_scripts
creating build/bdist.linux-aarch64/egg/EGG-INFO/scripts
copying build/scripts-3.7/minimap2.py -> build/bdist.linux-aarch64/egg/EGG-INFO/scripts
changing mode of build/bdist.linux-aarch64/egg/EGG-INFO/scripts/minimap2.py to 755
copying mappy.egg-info/PKG-INFO -> build/bdist.linux-aarch64/egg/EGG-INFO
copying mappy.egg-info/SOURCES.txt -> build/bdist.linux-aarch64/egg/EGG-INFO
copying mappy.egg-info/dependency_links.txt -> build/bdist.linux-aarch64/egg/EGG-INFO
copying mappy.egg-info/top_level.txt -> build/bdist.linux-aarch64/egg/EGG-INFO
writing build/bdist.linux-aarch64/egg/EGG-INFO/native_libs.txt
zip_safe flag not set; analyzing archive contents...
__pycache__.mappy.cpython-37: module references __file__
creating dist
creating 'dist/mappy-2.14-py3.7-linux-aarch64.egg' and adding 'build/bdist.linux-aarch64/egg' to it
removing 'build/bdist.linux-aarch64/egg' (and everything under it)
Creating /home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/minimap2-2.14-st347n5vburvgv2qijtmmsfa2w2oteno/lib/python3.7/site-packages/site.py
Processing mappy-2.14-py3.7-linux-aarch64.egg
creating /home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/minimap2-2.14-st347n5vburvgv2qijtmmsfa2w2oteno/lib/python3.7/site-packages/mappy-2.14-py3.7-linux-aarch64.egg
Extracting mappy-2.14-py3.7-linux-aarch64.egg to /home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/minimap2-2.14-st347n5vburvgv2qijtmmsfa2w2oteno/lib/python3.7/site-packages
Adding mappy 2.14 to easy-install.pth file
Installing minimap2.py script to /home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/minimap2-2.14-st347n5vburvgv2qijtmmsfa2w2oteno/bin

Installed /home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/minimap2-2.14-st347n5vburvgv2qijtmmsfa2w2oteno/lib/python3.7/site-packages/mappy-2.14-py3.7-linux-aarch64.egg
Processing dependencies for mappy==2.14
Finished processing dependencies for mappy==2.14
==> [2020-06-16-06:18:55.191017, 2237] 'make' '-j4' 'arm_neon=1' 'aarch64~1'
make: *** No rule to make target 'aarch64~1'.  Stop.
==> Error: ProcessError: Command exited with status 2:
    'make' '-j4' 'arm_neon=1' 'aarch64~1'

1 error found in build log:
     135    Installing minimap2.py script to /home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/minimap2-2.14-st347n5vburvgv2qijtmmsfa2w2oteno/bin
     136
     137    Installed /home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/minimap2-2.14-st347n5vburvgv2qijtmmsfa2w2oteno/lib/python3.7/site-packages/mappy-2.14-py3.7-linux-aarch64.egg
     138    Processing dependencies for mappy==2.14
     139    Finished processing dependencies for mappy==2.14
     140    ==> [2020-06-16-06:18:55.191017, 2237] 'make' '-j4' 'arm_neon=1' 'aarch64~1'
  >> 141    make: *** No rule to make target 'aarch64~1'.  Stop.

See build log for details:
  /tmp/root/spack-stage/spack-stage-minimap2-2.14-st347n5vburvgv2qijtmmsfa2w2oteno/spack-build-out.txt

==> Error: Failed to install minimap2 due to ChildError: ProcessError: Command exited with status 2:
    'make' '-j4' 'arm_neon=1' 'aarch64~1'
1 error found in build log:
     135    Installing minimap2.py script to /home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/minimap2-2.14-st347n5vburvgv2qijtmmsfa2w2oteno/bin
     136
     137    Installed /home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/minimap2-2.14-st347n5vburvgv2qijtmmsfa2w2oteno/lib/python3.7/site-packages/mappy-2.14-py3.7-linux-aarch64.egg
     138    Processing dependencies for mappy==2.14
     139    Finished processing dependencies for mappy==2.14
     140    ==> [2020-06-16-06:18:55.191017, 2237] 'make' '-j4' 'arm_neon=1' 'aarch64~1'
  >> 141    make: *** No rule to make target 'aarch64~1'.  Stop.

See build log for details:
  /tmp/root/spack-stage/spack-stage-minimap2-2.14-st347n5vburvgv2qijtmmsfa2w2oteno/spack-build-out.txt
Traceback (most recent call last):
  File "/home/xiaojun/spack/lib/spack/spack/build_environment.py", line 813, in child_process
    return_value = function()
  File "/home/xiaojun/spack/lib/spack/spack/installer.py", line 1123, in build_process
    phase(pkg.spec, pkg.prefix)
  File "/home/xiaojun/spack/lib/spack/spack/package.py", line 112, in phase_wrapper
    callback(instance)
  File "/home/xiaojun/spack/var/spack/repos/builtin/packages/minimap2/package.py", line 34, in install_minimap2
    mkdirp(prefix.bin)
  File "/home/xiaojun/spack/lib/spack/spack/build_environment.py", line 131, in __call__
    return super(MakeExecutable, self).__call__(*args, **kwargs)
  File "/home/xiaojun/spack/lib/spack/spack/util/executable.py", line 189, in __call__
    proc.returncode, long_msg)
spack.util.executable.ProcessError: Command exited with status 2:
    'make' '-j4' 'arm_neon=1' 'aarch64~1'
```

--- a/var/spack/repos/builtin/packages/minimap2/package.py
+++ b/var/spack/repos/builtin/packages/minimap2/package.py
@@ -28,7 +28,7 @@ class Minimap2(PythonPackage):
         if self.spec.target.family == 'aarch64':
             make_arg.extend([
                 'arm_neon=1',
-                'aarch64~1'
+                'aarch64=1'
             ])
         make(*make_arg)
         mkdirp(prefix.bin)

